### PR TITLE
feat(add budget method): implement add budget method

### PR DESCRIPTION
**Pull request by HoangLiarPhuongLe**
- Save Budget in Database
- Render Budget List

### CODE REVIEW

File: src/javascripts/controllers/transactionController.js [hunk 1]
```diff
@@ -66,27 +67,60 @@ class TransactionController {
...
+     */
+    async initBudget() {
+        await this.service.budgetService.initBudgetList();
+        this.loadListBudgets();
+        this.view.homeView.addEventRenderBudgetPopup(this.addBudget);
     }
+
+    loadListBudgets = async () => {
+        const budgets = this.service.budgetService.getBudgets();
+        const category = 'Other Income';
```
Comment (phuocduong-agilityio) [inline src/javascripts/controllers/transactionController.js]: Could you use another name that make sense?
After const, should have empty line

File: src/javascripts/templates/budgetTemplate.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    static createBudget = (budget) => {
+        const budgetDateTransform = formatDate(budget.date);
+        const budgetDate = budgetDateTransform.split(' ');
+
+        return `
+          <li class="transaction-detail">
+            <div class="transaction-info">
+              <div class="datetime">
+                <p class="date-text">${new Date(budget.date).getDate()}</p>
+                <div>
```
Comment (phuocduong-agilityio) [inline src/javascripts/templates/budgetTemplate.js]: WHat is this div use for?

File: src/javascripts/views/budgetView.js [hunk 3]
```diff
@@ -12,6 +13,19 @@ class BudgetView {
...
+    closeBudgetPopup() {
+        this.budgetElement.classList.remove('budget-active');
+        this.overlayElement.classList.remove('overlay-active');
+    }
+
+    resetBudgetPopup() {
+        const inputs = this.budgetBodyElement.querySelectorAll('input');
+
+        inputs.forEach((input) => {
+            input.value = ' ';
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/budgetView.js:25]: Have you try reset() for form?

Comment (HoangLiarPhuongLe) [reply src/javascripts/views/budgetView.js:25]: Yes, I have tried reset() for form @phuocduong-agilityio

File: src/javascripts/controllers/transactionController.js (hunk 1 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/controllers/transactionController.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/templates/budgetTemplate.js (hunk 2 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/templates/budgetTemplate.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/views/budgetView.js (hunk 3 above)
Comment (phuocduong-agilityio) [reply src/javascripts/views/budgetView.js:25]: @HoangLiarPhuongLe If form.reset is not work, you can use this way. However I think it should be ''

**Pull request merged**